### PR TITLE
[LSP] Return minimal edits for code actions when possible

### PR DESCRIPTION
**Pull request by allisonchou**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1223716/

Normal documents have a `GetTextChangesAsync` method, which is much more efficient than the previously used `TextDocument.GetTextChanges` method as it returns minimal edits instead of dirtying the entire file.

### CODE REVIEW

File: src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs [hunk 1]
```diff
@@ -170,19 +171,30 @@ public async Task<LSP.VSCodeAction> HandleRequestAsync(LSP.VSCodeAction codeActi
...
+                    // method instead, we would get a change that spans the entire document, which we ideally want to avoid.
+                    if (newTextDoc is Document newDoc && oldTextDoc is Document oldDoc)
+                    {
+                        textChanges = await newDoc.GetTextChangesAsync(oldDoc, cancellationToken).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        var newText = await newTextDoc.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                        textChanges = newText.GetTextChanges(oldText);
+                    }
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs:194]: we have a IDocumentTextDifferencingService.  can you investigate if we should be using that instead?

Comment (CyrusNajmabadi) [reply src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs:194]: actually, i'm fairly certain you should use that.  That service will fall bcak to this logic, but can use even better logic in a host that supports it.

**Review by CyrusNajmabadi — APPROVED** [review]
LGTM, but would recommend using this other service.

**Review by davidwengier — APPROVED** [review]

**Review by CyrusNajmabadi — APPROVED** [review]

**Review by davidwengier — APPROVED** [review]

**Comment by allisonchou** [comment]
Thanks Cyrus and David!

**Pull request merged**